pkg/plugin/utils/volumehelper/volume_policy_helper.go

```diff
@@ -60,3 +92,92 @@ func ShouldPerformSnapshotWithBackup(
 
 	return volumeHelperImpl.ShouldPerformSnapshot(unstructured, groupResource)
 }
+
+// NewVolumeHelperForBackup creates a VolumeHelper for the given backup with a PVC-to-Pod cache.
+// The cache is built for the provided namespaces list to avoid O(N*M) complexity when there
+// are many PVCs and pods. See issue #9179 for details.
+//
+// This function is intended for BIA plugins to create a VolumeHelper once and reuse it
+// across multiple Execute() calls for the same backup.
+//
+// If namespaces is nil or empty, the function will resolve the namespace list from the backup spec.
+// If backup.Spec.IncludedNamespaces is empty (meaning all namespaces), it will list all namespaces
+// from the cluster.
+func NewVolumeHelperForBackup(
```

Comment:
I don't see this function is used, can we remove it if not?
From the comments of this function, it seems to be used by plugins. However, plugins don't have enough information to provide the correct namespace list in the backup, the backup may have multiple ways to opt-in/opt-out namespaces. Regarding to this, we should not provide a function like this to plugins.